zipline/assets/assets.py

```diff
@@ -175,6 +175,10 @@ def _retrieve_futures_contract(self, sid):
     def _select_asset_by_sid(asset_tbl, sid):
         return sa.select([asset_tbl]).where(asset_tbl.c.sid == int(sid))
 
+    @staticmethod
+    def _select_asset_by_symbol(asset_tbl, symbol):
+        return sa.select([asset_tbl]).where(asset_tbl.c.symbol == symbol)
```

Comment:
It will result in multiple rows. This won't happen internally, as our futures symbols are unique. Additionally, we enforce a `UNIQUE` constraint on symbols for futures, so if this would only ever return multiple rows if 1) a user had removed this constraint and 2) the user's data contained duplicate symbols. In such a case it would probably make more sense for the user to change the root symbol of one of the duplicates.